app/components/post_draft/quick_actions/input_quick_action/index.tsx

```diff
@@ -34,28 +41,78 @@ export default function InputQuickAction({
     testID,
     disabled,
     inputType,
+    cursorPosition,
     updateValue,
+    updateCursorPosition,
     focus,
 }: Props) {
     const theme = useTheme();
+    const intl = useIntl();
     const onPress = useCallback(() => {
+        if (inputType === 'at') {
+            updateValue((v) => {
+                const part1 = v.substring(0, cursorPosition) + '@';
+                const part2 = v.substring(cursorPosition);
+
+                const value = part1 + part2;
+
+                updateCursorPosition(part1.length);
+                return value;
+            });
+            focus();
+        } else if (inputType === 'slash') {
+            updateValue((v) => `/${v}`);
+            focus();
+        } else {
+            openEmojiPicker();
+        }
+    }, [inputType, cursorPosition, updateValue, updateCursorPosition]);
+
+    const handleEmojiClick = useCallback((emoji: string) => {
         updateValue((v) => {
-            if (inputType === 'at') {
-                return `${v}@`;
+            let emojiDraft: string;
+            const {emojiCode, emojiData} = getEmojiCodeAndData(emoji, []);
+
+            if (emojiData?.image && emojiData.category !== 'custom') {
```

Comment:
We already have this if inside `getEmojiCodeAndData` so I wonder if there is any "readable and nice" way to get rid of it.

I understand we still need to know if it is a unicode emoji or just text to add or not the `:`, and that we don't add the `:` consistently in the two places this function is called, so... Happy to leave it as it is, but if you come up with a cleaner way, I am all ears :smile:

Reply:
ok, I'll try